### DISCUSSION TITLE
Subtraction Before & After Row

### DISCUSSION THREAD

**Discussion by muksidin** [Budibase/budibase#8347] [Help] [answered]
Hello Expert,
![Screenshot_20221023_022957](https://user-images.githubusercontent.com/5665115/197359159-fbeb612d-b94b-490e-bd98-2a8f931073d7.png)
?
How to subtraction before & after row in one column and the result in other column
I use JS, but not working.  Here is my JS script, but the result not okay.
`if (currentValue){
	var nextValue=currentValue;
	return nextValue;
}
var currentValue=$('All Total');
if (nextValue){
	return (currentValue-nextValue)*1000;
}`
May be any one help me to write the coding. Thanks.

**Comment by melohagan** [comment, marked as answer]
Hey @muksidin 

Just to clarify, are you using Formulas? https://docs.budibase.com/docs/formula

**Reply by muksidin** [reply]
@melohagan i use JS because i can not found function in Formulas to substract row in one column. Before i use Logmedo Application and use JS like my post & it work. 
Can you help me to write the JS code or may be there is a function in Formulas which i don't know.
Thanks you

**Reply by melohagan** [reply]
I think in this case, Formulas wouldn't work because they allow for calculation between columns of the same row, whereas you want to perform calculations between rows. 

If you're using an external db then you could try a custom query, which I imagine would be fairly complicated. 

Otherwise you could use try using JavaScript on the front-end Data provider but it's probably not ideal.
Something like:
```
let rows = $("New Data Provider.Rows");
for (let row of rows) {
  //TO-DO
}
```
This method will only support a single page of data provider results.

**Reply by muksidin** [reply]
Can you write complete in `//TO-DO` script to substruct row 2-row1 etc.

**Reply by melohagan** [reply]
This StackOverflow post may help: https://stackoverflow.com/questions/20477177/creating-an-array-of-cumulative-sum-in-javascript#55261098

In this case the array is `rows`, and you wouldn't need the for loop.

**Reply by muksidin** [reply]
Thanks alot @melohagan ,may be this function (substract for after with before) can add in budibase feature.

**Comment by muksidin** [comment]
Hello @melohagan , can you help me to solve this. I attach the export application. I want to substract row 2-1, 3-2, 4-3 etc ... and result in column result.
[SubstractRow-export-1671389044360.tar.gz](https://github.com/Budibase/budibase/files/10254433/SubstractRow-export-1671389044360.tar.gz)

**Comment by melohagan** [comment]
Hey @muksidin 

As I had mentioned:
> Formulas wouldn't work because they allow for calculation between columns of the same row, whereas you want to perform calculations between rows.

I was able to create a cumulative sum in the **Design** section, however it will only work on the currently displayed page of data.

1. Add a Data Provider with Ascending sort on the Auto ID:
![Screenshot 2022-12-19 at 09 25 02](https://user-images.githubusercontent.com/101575380/208392289-0a2e4e38-2c1e-480a-a0ef-514e2e7fa934.png)

2. Add a Repeater with a Paragraph. Give the paragraph the following JavaScript binding:
```js
let sum = 0;
for (let i = 0; i < $("Repeater.Example.Auto ID"); i++) {
	sum += $("Dataprovider.Rows")[i].Data;
}
return sum;
```

**Result**
![Screenshot 2022-12-19 at 09 28 49](https://user-images.githubusercontent.com/101575380/208393055-ad2b9a9d-7425-4dd9-99f9-5466abb4c8fa.png)

**App export**
[Subtract Row-export-1671442148426.tar.gz](https://github.com/Budibase/budibase/files/10257966/Subtract.Row-export-1671442148426.tar.gz)

**Reply by melohagan** [reply]
Just realised I mis-read your request. Similar principle, just use the following JavaScript binding instead:
```js
let value = $("Dataprovider.Rows")[0]?.Data;
for (let i = 1; i < $("Repeater.Example.Auto ID"); i++) {
    value = $("Dataprovider.Rows")[i]?.Data - $("Dataprovider.Rows")[i-1]?.Data;
}
return value;
```

**Result**
![Screenshot 2022-12-19 at 09 34 37](https://user-images.githubusercontent.com/101575380/208394124-86be354d-e3dd-4aee-bc96-19566d6203de.png)

**Reply by muksidin** [reply]
Is it possible if the result in table?
in attachment i want the result is in `result` column.
